src/Components/CreateImageWizard/ValidatedInput.tsx

```diff
@@ -38,48 +47,89 @@ type HookValidatedInputPropTypes = TextInputProps &
     inputType?: 'textInput' | 'textArea';
   };
 
-export const HookPasswordValidatedInput = ({
-  ariaLabel,
-  placeholder,
-  dataTestId,
+type ValidatedPasswordInput = TextInputProps & {
+  value: string;
+  stepValidation: StepValidation;
+  fieldName: string;
+};
+
+type PasswordInputProps = TextInputProps & {
+  value: string;
+  errorMessage: string;
+};
+
+type ErrorMessageProps = {
+  errorMessage: string;
+};
+
+export const ValidatedPasswordInput = ({
   value,
-  ouiaId,
   stepValidation,
   fieldName,
   onChange,
-  warning = undefined,
-  inputType,
-  isDisabled,
-}: HookValidatedInputPropTypes) => {
+  onBlur,
+  placeholder,
+}: ValidatedPasswordInput) => {
+  const errorMessage = stepValidation.errors[fieldName];
+  const hasError = errorMessage !== '';
+
+  return (
+    <>
+      <PasswordInput
+        value={value}
+        placeholder={placeholder || ''}
+        errorMessage={errorMessage}
+        onChange={onChange!}
```

Comment:
Do we need the non-null assertion operator here? :thinking:

Reply:
because it detects that the properties onChange and onBlur might be undefined at runtime, even though the code seems to assume they are always defined. The reason is that they are defined as Optional props.